pip/index.py

```diff
@@ -215,7 +215,14 @@ def sort_path(path):
                 elif os.path.isfile(path):
                     sort_path(path)
             else:
-                urls.append(url)
+                scheme = urllib_parse.urlsplit(url).scheme
```

Comment:
Unfortunately:

    In [17]: urllib_parse.urlsplit('c:\qsdqs').scheme
    Out[17]: 'c'

I'm wondering whether we should whitelist a specific set of scheme ?

    ['file', 'http', 'https'] + pip.vcs.vcs.all_schemes